packages/grid/_modules_/grid/components/columnHeaders/ColumnHeaderItem.tsx

```diff
@@ -26,7 +26,7 @@ interface ColumnHeaderItemProps {
   filterItemsCounter?: number;
 }
 
-export const ColumnHeaderItem = ({
+export const ColumnHeaderItem = React.memo(({
```

Comment:
An important detail, we can't memoize components that depends on 

```jsx
const apiRef = React.useContext(ApiContext);
```

The logic isn't setup to allow to bypass the memoization when the state change and the component needs to rerender. So, in the current state of things, it's not an option. We have already faced this issue for:

- exposing a composable Toolbar API
- babel plugin that automatically hoists
- and similar cases like this one in the past, e.g. toolbar density menu item